elastic/kibana
# [Infra UI] Implement telemetry for the asset details flyout

## Summary

This PR adds proper `data-test-subj` attributes to the Asset Details component and adds a new custom telemetry event


<img width="623" alt="image" src="https://github.com/elastic/kibana/assets/2767137/0e637e1a-4447-43c7-9e19-86974dbb13c0">

<img width="623" alt="image" src="https://github.com/elastic/kibana/assets/2767137/881cd52d-8c94-49bb-a6cb-781293ec2ecb">



Besides that, I've renamed 2 props,`node` -> `asset`  and `nodeType` -> `assetType`, to make naming consistent with what we're naming these attributes across asset-related stuff.


### How to test

- Setup a local Kibana instance
- Navigate to `Infrastructure` > `Hosts`
- Open the flyout
  - Upon opening, an event called `Asset Details Flyout Viewed` should be present in `kibana-browser` request 
  - Click through the flyout and check the `kibana-browser` requests
    - All automatic events in the flyout should contain the `data-component-name` and `data-asset-type` attributes
